### PR TITLE
UI controller functions

### LINKED ISSUE #4: Create A helper method II
A helper method that retrives a existing time track.

### PR DESCRIPTION
**Pull request by boergerj**
Adds function to controller getTimingDetails which allows UI to get relevant timing information when wanting to view timing breakdown screen

Changes:
Adds getTimingDetails
Uses round in displayTimeDetail() call
Replaces placeholder in displayTimeDetail definition in time_canvas_handler and makes necessary adjustments to receive JSON

### CODE REVIEW

**Review by Hatsukikeika — APPROVED** [review]
Tested on local and it works fine

File: app/assets/javascripts/time_canvas_handler.js [hunk 1]
```diff
@@ -49,19 +49,23 @@ function displayTimeDetail(resp_map_id,round){
     $("#timeModalBody").empty().html("<p>Loading data...(This might need few seconds)</p>")
     $("#timeModal").show();
     $.ajax({
-        url: "Url_placeholder?response_map_id=" + resp_map_id + "&round=" + round ,
-        method: 'GET',
+        url: '/submission_viewing_events/getTimingDetails',
+        method: 'POST',
```
Comment (AvlWx2014) [inline app/assets/javascripts/time_canvas_handler.js:53]: Why are we using POST?

Comment (AvlWx2014) [reply app/assets/javascripts/time_canvas_handler.js:53]: Is there a request body?

File: app/controllers/submission_viewing_events_controller.rb [hunk 2]
```diff
@@ -88,6 +89,45 @@ def mark_end_time
...
+    require 'json'
+    labels = [] # store links accessed during review
+    percentages = [] # store percentages per link for pie chart
+    tables = [] # store timing data breakdown per link
+
+    # get total time spent on review
+    totalTime = getTotalTime(params[:reponse_map_id], params[:round])
+
+    # get all timing entries for review (each link has one entry)
+    timingEntries = SubmissionViewingEvent.where(map_id: params[:reponse_map_id], round: params[:round])
```
Comment (AvlWx2014) [inline app/controllers/submission_viewing_events_controller.rb:103]: Do we need to consider flushing LocalStorage before processing this?

**Review by AvlWx2014 — APPROVED** [review]

File: app/controllers/submission_viewing_events_controller.rb (hunk 2 above)
Comment (boergerj) [reply app/controllers/submission_viewing_events_controller.rb:103]: We could but this should never be called until the review is done so it should not matter

File: app/assets/javascripts/time_canvas_handler.js (hunk 1 above)
Comment (boergerj) [reply app/assets/javascripts/time_canvas_handler.js:53]: This is what my research said to do. It makes the post request then the json is rendered in return for the UI to use

**Pull request merged**